packages/grid/x-data-grid/src/hooks/core/strategyProcessing/useGridStrategyProcessing.ts

```diff
@@ -0,0 +1,133 @@
+ *    So listening to both would most likely run your logic twice.
+ * - `GridEvents.activeStrategyProcessorChange` to update something when the processor of the active strategy changes
+ *
+ * The active strategy is the first available strategy based on the registration order
+ * Which means that if two strategies are available, the active one will be the one of the hook ran 1st.
+ *
+ * For now, this hook is only compatible with row grouping related strategies.
+ * In the future, it could support several type of strategies if we wanted to apply the same pattern to another set of processors.
+ */
+export const useGridStrategyProcessing = (apiRef: React.MutableRefObject<GridApiCommunity>) => {
+  const availableStrategies = React.useRef(new Map<string, () => boolean>());
+  const strategiesCache = React.useRef<{
+    [P in GridStrategyProcessorName]?: { [strategyName: string]: GridStrategyProcessor<P> };
+  }>({});
+
+  const registerStrategyProcessor = React.useCallback<
+    GridStrategyProcessingApi['unstable_registerStrategyProcessor']
+  >(
+    (strategyName, group, processor: GridStrategyProcessor<any>) => {
+      if (!strategiesCache.current[group]) {
+        strategiesCache.current[group] = {};
+      }
+
+      const groupPreProcessors = strategiesCache.current[group]!;
+      const previousProcessor = groupPreProcessors[strategyName];
+      if (previousProcessor !== processor) {
+        groupPreProcessors[strategyName] = processor;
+
+        if (
+          (previousProcessor as GridStrategyProcessor<any> | undefined) &&
+          strategyName === apiRef.current.unstable_getActiveStrategy()
+        ) {
+          apiRef.current.publishEvent(GridEvents.activeStrategyProcessorChange, group);
+        }
+      }
+
+      return () => {
+        const { [strategyName]: removedPreProcessor, ...otherProcessors } =
+          strategiesCache.current[group]!;
+        strategiesCache.current[group] = otherProcessors as {
+          [strategyName: string]: GridStrategyProcessor<any>;
+        };
+      };
+    },
+    [apiRef],
+  );
+
+  const applyStrategyProcessor = React.useCallback<
+    GridStrategyProcessingApi['unstable_applyStrategyProcessor']
+  >(
+    (group, params) => {
+      const currentStrategy = apiRef.current.unstable_getActiveStrategy();
+      if (currentStrategy == null) {
+        throw new Error("Can't apply a strategy processor before defining an active strategy");
+      }
+
+      const groupCache = strategiesCache.current[group];
+      if (!groupCache || !groupCache[currentStrategy]) {
+        throw new Error(`No processor found for group "${group}" on strategy "${currentStrategy}"`);
+      }
+
+      const processor = groupCache[currentStrategy] as GridStrategyProcessor<any>;
+      return processor(params);
+    },
+    [apiRef],
+  );
+
+  const getActiveStrategy = React.useCallback<
+    GridStrategyProcessingApi['unstable_getActiveStrategy']
+  >(() => {
```

Comment:
Should it receive the group as argument and return which strategy is active for that specific group?

Currently this hook can only be used to group rows, but it may be useful for other things too.

Reply:
Done
I added a concept of `strategyGroup`, with only one existing right now (`rowTree` which contains the 3 pre-processors `rowTreeCreation`, `sorting`, `filtering`)
That way, if we want to create pre-proccessors with an independent list of strategies, we will just have to put it in another strategy group).

Everything is transparent, except for `getActiveStrategy` which takes the strategy group as a param, as you proposed.